pallets/creditcoin/src/ocw/tasks/collect_coins.rs

```diff
@@ -906,4 +915,37 @@ pub(crate) mod tests {
 			);
 		});
 	}
+
+	#[test]
+	fn persist_minimum_existencial_deposit_errors() {
+		let mut ext = ExtBuilder::default();
+		let acct_pubkey = ext.generate_authority();
+		let auth = AccountId::from(acct_pubkey.into_account().0);
+		ext.build_offchain_and_execute_with_state(|_, _| {
+			let (acc, addr, sign, _) = generate_address_with_proof("collector");
```

Comment:
I'm not following something here. Where does `acc` gets its balance from (which line of code) ? 

- Better to add an assert that the pre-condition for balance holds true in case we change something in the future.

- If the balance is under the existential minimum how does register_address() succeed below ? Isn't it going to fail b/c fees would not be able to be paid ?

Reply:
You can preset balances when you `assimilate_storage` like it happens in `mock.rs` a shorter example can be found in #428. More concretely, in this test, acc's balance doesn't exist yet. Fees and weights are out of scope for the mocked Externalities; we mainly assert on storage changes and txn pool insertion for the OCW. Extrinsic calls output a DispatchResult variant that the assert family macro matches on. 

acc's balance only matters (under this context) when an extrinsic call mutates its balance. Such a minting into its balance, e.g. persisting collected coins.

Reply:
Also, [this](https://github.com/gluwa/substrate/blob/master/frame/balances/src/lib.rs#L827) is the operator you were looking for.